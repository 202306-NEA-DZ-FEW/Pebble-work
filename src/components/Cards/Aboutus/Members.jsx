import React, { useState, useEffect } from "react";

// Import Swiper React components
import { Swiper, SwiperSlide } from "swiper/react";

// Import Swiper styles
import "swiper/css";
import "swiper/css/effect-coverflow";
import "swiper/css/pagination";

// import required modules
import {
    EffectCoverflow,
    Pagination,
    Autoplay,
    Navigation,
} from "swiper/modules";
import Link from "next/link";
import Image from "next/image";

export default function Members({ members }) {
    const [direction, setDirection] = useState("");

    useEffect(() => {
        const handleResize = () => {
            if (typeof window !== "undefined") {
                setDirection(
                    window.innerWidth < 640 ? "vertical" : "horizontal"
                );
            }
        };

        // Call handleResize once to set the direction based on the initial window size
        handleResize();

        if (typeof window !== "undefined") {
            window.addEventListener("resize", handleResize);
        }

        return () => {
            if (typeof window !== "undefined") {
                window.removeEventListener("resize", handleResize);
            }
        };
    }, []);

    return (
        <Swiper
            direction={direction}
            effect={"coverflow"}
            grabCursor={true}
            centeredSlides={true}
            slidesPerView={2}
            loop={true}
            autoplay={{
                delay: 5000,
                disableOnInteraction: false,
            }}
            coverflowEffect={{
                rotate: 70,
                stretch: 0,
                depth: 100,
                modifier: 1,
                slideShadows: false,
            }}
            pagination={true}
            modules={[EffectCoverflow, Pagination, Autoplay, Navigation]}
        >
            {members.map((member, index) => (
                <SwiperSlide key={index}>
                    <div className='h-full shadow-md w-full flex flex-col items-center justify-evenly'>
                        <img
                            src={member.imageSrc}
                            alt='member'
                            className='rounded-full w-[30vw] sm:w-[80px] lg:w-[140px] '
                        />
                        <div className='flex flex-col items-center'>
                            <h2 className='text-black font-[500]'>
                                {member.name}
                            </h2>
                            <p className='slide-content p-5 text-black'>
                                {member.description}
                            </p>
                            <div className='flex w-full justify-evenly slide-content'>
                                <Link
                                    href={member.google}
                                    key='index'
                                    className='social-media-icon'
                                >
                                    <Image
                                        className='xl:w-[50px] xl:h-[50px] md:w-[40px] md:h-[40px] sm:w-[30px] sm:h-[30px] w-[22px] h-[22px]'
                                        width={50}
                                        height={50}
                                        src='/icons/Google.svg'
                                        alt='Gmail'
                                    />
                                </Link>
                                <Link
                                    href={member.linkedin}
                                    key='index'
                                    className='social-media-icon'
                                >
                                    <Image
                                        className='xl:w-[50px] xl:h-[50px] md:w-[40px] md:h-[40px] sm:w-[30px] sm:h-[30px] w-[22px] h-[22px]'
                                        width={50}
                                        height={50}
                                        src='/icons/Linkedin.png'
                                        alt='Linkedin'
                                    />
                                </Link>
                                <Link
                                    href={member.github}
                                    key='index'
                                    className='social-media-icon'
                                >
                                    <Image
                                        className='xl:w-[50px] xl:h-[50px] md:w-[40px] md:h-[40px] sm:w-[30px] sm:h-[30px] w-[22px] h-[22px]'
                                        width={50}
                                        height={50}
                                        src='/icons/Github.svg'
                                        alt='Github'
                                    />
                                </Link>
                            </div>
                        </div>
                    </div>
                </SwiperSlide>
            ))}
        </Swiper>
    );
}
