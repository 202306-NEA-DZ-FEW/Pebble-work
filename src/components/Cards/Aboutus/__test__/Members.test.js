jest.mock("swiper/modules", () => ({
    EffectCoverflow: jest.fn(),
    Pagination: jest.fn(),
    Autoplay: jest.fn(),
    Navigation: jest.fn(),
}));
jest.mock("swiper/react", () => ({
    Swiper: ({ children }) => <div>Mock Swiper {children}</div>,
    SwiperSlide: ({ children }) => <div>Mock SwiperSlide {children}</div>,
}));

import React from "react";
import renderer from "react-test-renderer";
import Members from "../Members";

describe("Test Component", () => {
    it("renders correctly", () => {
        const members = [
            {
                imageSrc: "/path/to/image.jpg",
                name: "Test Name",
                description: "Test Description",
                google: "https://google.com",
                linkedin: "https://linkedin.com",
                github: "https://github.com",
            },
            // add more member objects as needed
        ];

        const tree = renderer.create(<Members members={members} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
