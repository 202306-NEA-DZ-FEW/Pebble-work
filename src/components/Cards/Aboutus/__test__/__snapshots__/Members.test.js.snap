// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component renders correctly 1`] = `
<div>
  Mock Swiper 
  <div>
    Mock SwiperSlide 
    <div
      className="h-full w-full flex flex-col items-center justify-evenly"
    >
      <img
        alt="member"
        className="rounded-full w-[30vw] sm:w-[80px] lg:w-[140px] "
        src="/path/to/image.jpg"
      />
      <div
        className="flex flex-col items-center"
      >
        <h2
          className="text-red-600"
        >
          Test Name
        </h2>
        <p
          className="slide-content p-5 text-black"
        >
          Test Description
        </p>
        <div
          className="flex w-full justify-evenly slide-content"
        >
          <a
            className="social-media-icon"
            href="https://google.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <img
              alt="Gmail"
              className="xl:w-[50px] xl:h-[50px] md:w-[40px] md:h-[40px] sm:w-[30px] sm:h-[30px] w-[22px] h-[22px]"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/icons/Google.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
          <a
            className="social-media-icon"
            href="https://linkedin.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <img
              alt="Linkedin"
              className="xl:w-[50px] xl:h-[50px] md:w-[40px] md:h-[40px] sm:w-[30px] sm:h-[30px] w-[22px] h-[22px]"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Ficons%2FLinkedin.png&w=128&q=75"
              srcSet="/_next/image?url=%2Ficons%2FLinkedin.png&w=64&q=75 1x, /_next/image?url=%2Ficons%2FLinkedin.png&w=128&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
          <a
            className="social-media-icon"
            href="https://github.com"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <img
              alt="Github"
              className="xl:w-[50px] xl:h-[50px] md:w-[40px] md:h-[40px] sm:w-[30px] sm:h-[30px] w-[22px] h-[22px]"
              data-nimg="1"
              decoding="async"
              height={50}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/icons/Github.svg"
              style={
                {
                  "color": "transparent",
                }
              }
              width={50}
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
